html/js/guiutils/ContentEditable.js

```diff
@@ -0,0 +1,181 @@
+        this.textdiv.addEventListener('input', function(e) {
+            // Set the new width based on the text width
+            newWidth = thisContentEditable.textdiv.scrollWidth +
+                            GTE.CONSTANTS.CONTENT_EDITABLE_FOREIGN_EXTRA_WIDTH;
+            thisContentEditable.myforeign.setAttribute("width", newWidth);
+            if (growingOfText === GTE.CONSTANTS.CONTENT_EDITABLE_GROW_TO_LEFT) {
+                // Calculate how much has the foreign grown
+                x -= newWidth - previousWidth;
+                // Translate the foreign object that amount to the left
+                thisContentEditable.myforeign.setAttributeNS(null, "transform", "translate(" + x + " " + y + ")");
+            }
+            previousWidth = newWidth;
+        });
+
+        // blur event is used to detect when the contenteditable loses focus
+        this.textdiv.addEventListener('blur', function(e) {
+            if (thisContentEditable.functionOnSave !== null){
+                // Save the object specifi on functionOnSave
+                thisContentEditable.functionOnSave();
+            }
+        });
+
+        // keypress is used to detect the RETURN key and to apply a max number
+        // of characters allowed
+        this.textdiv.addEventListener('keypress', function(e) {
+            var max = 30;
+            // TODO #21
+            // Check for max number of chars
+            if(e.which != 8 && this.innerHTML.length > max) {
+               e.preventDefault();
+            }
+            if(e.which == 13) {
+                // If RETURN is pressed
+                // Blur the input instead of adding a line break
+                e.preventDefault();
+                this.blur();
+                window.getSelection().removeAllRanges();
+                return false;
+            }
+        });
+        return this;
+    }
+
+    /**
+    * Function that shows the foreign object by modifying its CSS display property
+    * @return {ContentEditable} this Returns this instance
+    */
+    ContentEditable.prototype.show = function () {
+        this.myforeign.style.display = "block";
+        return this;
+    };
+
+    /**
+    * Function that hides the foreign object by modifying its CSS display property
+    * @return {ContentEditable} this Returns this instance
+    */
+    ContentEditable.prototype.hide = function () {
+        this.myforeign.style.display = "none";
+        return this;
+    };
+
+    /**
+    * Function that returns current object visibility
+    * @return {Boolean} visible Returns true if object is visible
+    */
+    ContentEditable.prototype.visible = function () {
+        if (this.myforeign.style.display === "none") {
+            return false;
+        } else {
+            return true;
+        }
+    };
+
+    /**
+    * Function that specifies the function to be run on save (i.e. when TAB or
+    * RETURN key is pressed). It saves the function pointer specified on params
```

Comment:
Contradicting this comment, I think you aren't checking for TAB, and however you are checking for the out of focus (blur) event.